### PR TITLE
[SNOW-1649176]: Fix `inplace` argument for Series objects derived from DataFrame columns.

### PR DESCRIPTION
**Pull request by sfc-gh-rdurrani**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1649176

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

File: src/snowflake/snowpark/modin/pandas/indexing.py [hunk 1]
```diff
@@ -907,7 +907,11 @@ def __getitem__(
         )
         if isinstance(result, Series):
             result._parent = self.df
-            result._parent_axis = 0
+            # We need to determine which axis this Series was extracted from. We can do so
```
[the remark this answers is not in the document]
Comment (sfc-gh-rdurrani) [reply src/snowflake/snowpark/modin/pandas/indexing.py:910]: This updates `iloc` - as for `at` and `iat`, they didn't seem to set `_parent_axis`, but I'll double check!

[the remark this answers is not in the document]
Comment (sfc-gh-rdurrani) [reply src/snowflake/snowpark/modin/pandas/indexing.py:910]: `at` and `iat` return scalar values (single value lookup), so should not be affected by this!

[the remark this answers is not in the document]
Comment (sfc-gh-rdurrani) [reply src/snowflake/snowpark/modin/pandas/indexing.py:910]: It updates both? The second diff is for iloc and the first is for loc?

**Review by sfc-gh-azhan — APPROVED** [review]

**Review by sfc-gh-vbudati — APPROVED** [review]

**Review by sfc-gh-vbudati — APPROVED** [review]

File: src/snowflake/snowpark/modin/pandas/indexing.py [hunk 2]
```diff
@@ -1181,7 +1189,15 @@ def __getitem__(
 
         if isinstance(result, Series):
             result._parent = self.df
-            result._parent_axis = 0
+            # We need to determine which axis this Series was extracted from. We can do so
```
Comment (sfc-gh-evandenberg) [inline src/snowflake/snowpark/modin/pandas/indexing.py:1192]: This looks like redundant code with above, would it make sense to put this in a separate helper function?

**Review by sfc-gh-evandenberg — APPROVED** [review]
Small comment, otherwise LGTM

**Pull request merged**